front50-web/src/main/java/com/netflix/spinnaker/front50/controllers/PipelineController.java

```diff
@@ -237,33 +258,106 @@ public Pipeline getByApplicationAndName(
   }
 
   @PreAuthorize(
-      "@fiatPermissionEvaluator.storeWholePermission() and hasPermission(#pipeline.application, 'APPLICATION', 'WRITE') and @authorizationSupport.hasRunAsUserPermission(#pipeline)")
+      "@fiatPermissionEvaluator.storeWholePermission() "
+          + "and hasPermission(#pipeline.application, 'APPLICATION', 'WRITE') "
+          + "and @authorizationSupport.hasRunAsUserPermission(#pipeline)")
   @RequestMapping(value = "", method = RequestMethod.POST)
   public synchronized Pipeline save(
       @RequestBody Pipeline pipeline,
       @RequestParam(value = "staleCheck", required = false, defaultValue = "false")
           Boolean staleCheck) {
 
-    validatePipeline(pipeline, staleCheck);
+    long saveStartTime = System.currentTimeMillis();
+    log.info(
+        "Received request to save pipeline {} in application {}",
+        pipeline.getName(),
+        pipeline.getApplication());
 
+    log.debug("Running validation before saving pipeline {}", pipeline.getName());
+    long validationStartTime = System.currentTimeMillis();
+    validatePipeline(pipeline, staleCheck);
+    checkForDuplicatePipeline(
```

Comment:
Minor:  Should this be part of validation? It's done on both save & batch & would reduce the duplicate call I'd think...

Reply:
It's not called under batch. The way duplicates are checked is different for save and batch hence the common code is moved to validation and this method is kept outside.

Reply:
Ah k then I mis-read, thought it was duplicated between the two (the check for duplicate)